File: fapi/src/main/java/com/netcracker/ivanmerkush/fapi/service/CommentService.java
package com.netcracker.ivanmerkush.fapi.service;

import com.netcracker.ivanmerkush.fapi.models.Comment;
import com.netcracker.ivanmerkush.fapi.models.CommentViewModel;

import java.util.List;

public interface CommentService {
    Comment saveComment(Comment comment);
    void deleteComment(Integer id);
    List<CommentViewModel> getCommentsByIdPost(Integer idPost);
}

File: frontend/src/app/services/comment.service.ts
import {Injectable} from "@angular/core";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Comment} from "../models/commentModel";
import {CommentViewModel} from "../models/commentViewModel";

@Injectable()
export class CommentService {
  constructor(private http: HttpClient) {

  }

  getCommentsForPost(idPost: number): Observable<CommentViewModel[]> {
    return this.http.get<CommentViewModel[]>('/api/comments/' + idPost);
  }

  saveComment(comment: Comment): Observable<Comment> {
    return this.http.post<Comment>('/api/comments', comment);
  }

  deleteComment(idComment: number): Observable<void> {
    return this.http.delete<void>('/api/comments/' + idComment);
  }
}
